pkg/test/utils/kubernetes.go

```diff
@@ -372,16 +372,58 @@ func Namespaced(dClient discovery.DiscoveryInterface, obj runtime.Object, namesp
 	return m.GetName(), namespace, nil
 }
 
+func pruneLargeAdditions(expected *unstructured.Unstructured, actual *unstructured.Unstructured) runtime.Object {
+	pruned := actual.DeepCopy()
+	prune(expected.Object, pruned.Object)
+	return pruned
+}
+
+func prune(expected map[string]interface{}, actual map[string]interface{}) {
```

Comment:
seems worth a godoc to explain some of the logic or the purpose.  It's easy to understand based on name that we are "pruning" but pruning what and why and perhaps some how